### DISCUSSION TITLE
Given the javascript/editor representation of a block, is there a way to get it's rendered html?

### DISCUSSION THREAD

**Discussion by jakeparis** [WordPress/gutenberg#63581] [Q&A] [answered]
If I have in my component, the representation of a block like:
```js
{
  "clientId": "721d0ede-28d7-4e06-b430-18b594295865",
  "name": "core/paragraph",
  "isValid": true,
  "attributes": {
    "content": {},
    "dropCap": false
  },
  "innerBlocks": []
}
```
is there any function or method to transform this into the rendered html? 

For example, I would like to create a block that allows you use it's InnerBlocks to make snippets to save off to the database individually.

**Comment by talldan** [comment, marked as answer]
`wp.blocks.serialize( blocks )` should do the job.

If you're running this programmatically (outside of the context of an editor), then I believe you'd also have to register the block types first (`wp.blocks.registerBlockType` directly or by loading the block-library).

**Reply by jakeparis** [reply]
Perfect. Thank you so much, this is precisely what I needed!
